pyrk/th_system.py

```diff
@@ -53,6 +56,52 @@ def comp_from_name(self, name):
         msg += name
         raise KeyError(msg)
 
+
+
+class THSystemSphPS(THSystem):
+
+    def __init__(self, kappa, components):
+        THSystem.__init__(self, kappa, components)
+
+    def dtempdt(self, component, power, omegas, t_idx):
+        to_ret = 0*units.kelvin/units.second
+        cap = (component.rho(t_idx)*component.cp*component.vol)
+        if component.heatgen:
+            to_ret += self.heatgen(component, power, omegas)/cap
+        for name, d in component.adv.iteritems():
+            Qadv = self.advection(t_out=component.T[t_idx]*2.0 - d['t_in'],
+                                  t_in=d['t_in'],
+                                  m_flow=d['m_flow'],
+                                  cp=d['cp'])
+            to_ret -= Qadv/cap
+            if Qadv.magnitude < 0:
+                print '''at step %d, %s is heating the system by %f watts???
+            Tin is %f, tout is %f, tcomp is %f''' % (
+                    t_idx, component.name, Qadv.magnitude, d['t_in'].magnitude,
+                    (component.T[t_idx]*2 - d['t_in']).magnitude, component.T[t_idx].magnitude)
+        for interface, d in component.cond.iteritems():
+            env = self.comp_from_name(interface)
+            Qcond = self.conduction(t_b=component.T[t_idx],
+                                    t_env=env.T[t_idx],
+                                    r_b=d['r_b'],
+                                    r_env=d['r_env'],
+                                    k=d['k'])
+            to_ret -= Qcond/cap
+            assert (Qcond*(component.T[t_idx]-env.T[t_idx])).magnitude >= 0, 'conduction from %s to %s, from temp %f to %f is wrong %f' % (
+                component.name, env.name, component.T[t_idx].magnitude,
+                env.T[t_idx].magnitude, Qcond.magnitude)
+        for interface, d in component.conv.iteritems():
+            env = self.comp_from_name(interface)
+            Qconv = self.convection(t_b=component.T[t_idx],
+                                    t_env=env.T[t_idx],
+                                    h=d['h'],
+                                    A=d['area'])
+            to_ret -= Qconv/cap
+            assert (Qconv*(component.T[t_idx]-env.T[t_idx])).magnitude >= 0, 'cnvection from %s to %s, from temp %f to %f is wrong %f' % (
```

Comment:
convection is missing an o. Also, "is wrong" isn't a very informative error message.